remoulade/api/main.py

```diff
@@ -0,0 +1,42 @@
+""" This file describe the API to get the state of messages """
+from flask import Flask, jsonify, make_response, request
+from werkzeug.exceptions import BadRequest, NotFound
+
+import remoulade
+from remoulade.state import StateNamesEnum
+
+app = Flask(__name__)
+
+
+@app.route("/messages/states/")
+def get_states():
+    name_state = request.args.get("name")
+    backend = remoulade.get_broker().get_state_backend()
+    try:
+        data = None
+        if name_state is None:
+            data = [d for d in backend.get_n_states()]
+        else:
+            if not any(state_enum for state_enum in StateNamesEnum if state_enum.name.lower() == name_state.lower()):
+                raise BadRequest("{} state is not defined.".format(name_state))
+            data = [state.asdict() for state in backend.get_n_states() if state.name.lower() == name_state.lower()]
+        return make_response(jsonify(data), 200)
+    except BadRequest as e:
```

Comment:
do not catch the error here but in a error handler like this 
```
@app.errorhandler(HTTPException)
def http_exception(e):
    return {"error": str(e)}, e.code
```

as HTTPException is a superclass of BadRequest and NotFound, we will send a nice json when we get these exceptions